client/my-sites/earn/ads/wrapper.jsx

```diff
@@ -284,101 +296,47 @@ class AdsWrapper extends Component {
 				{ isAllowedSection ? component : <FeatureExample>{ component }</FeatureExample> }
 			</>
 		);
-	}
-
-	render() {
-		const {
-			canAccessAds,
-			canActivateWordadsInstant,
-			canUpgradeToUseWordAds,
-			isEnrolledWithIneligiblePlan,
-			isWordadsInstantEligibleButNotOwner,
-			site,
-			siteId,
-			translate,
-		} = this.props;
+	};
 
-		let component = this.props.children;
+	const getComponentAndNotice = () => {
+		let component = children;
 		let notice = null;
 
-		if ( this.props.requestingWordAdsApproval || this.props.wordAdsSuccess ) {
+		if ( requestingWordAdsApproval || wordAdsSuccess ) {
 			notice = (
 				<Notice status="is-success" showDismiss={ false }>
 					{ translate( 'You have joined the WordAds program. Please review these settings:' ) }
 				</Notice>
 			);
 		} else if ( canActivateWordadsInstant ) {
-			component = this.renderInstantActivationToggle( component );
+			component = renderInstantActivationToggle( component );
 		} else if ( isWordadsInstantEligibleButNotOwner ) {
-			component = this.renderOwnerRequiredMessage( component );
+			component = renderOwnerRequiredMessage( component );
 		} else if ( canUpgradeToUseWordAds && site.jetpack ) {
-			component = this.renderjetpackUpsell();
+			component = renderjetpackUpsell();
 		} else if ( canUpgradeToUseWordAds ) {
-			component = this.renderUpsell();
+			component = renderUpsell();
 		} else if ( ! canAccessAds ) {
-			component = this.renderEmptyContent();
+			component = renderEmptyContent();
 		} else if ( ! site.options.wordads && ! ( site.jetpack && canUpgradeToUseWordAds ) ) {
 			component = null;
 		} else if ( site.options.wordads && site.is_private ) {
-			notice = this.renderNoticeSiteIsPrivate();
+			notice = renderNoticeSiteIsPrivate();
 		} else if ( isEnrolledWithIneligiblePlan ) {
-			component = this.renderContentWithUpsell( component );
+			component = renderContentWithUpsell( component );
 		}
-
-		return (
-			<div>
-				<QuerySites siteId={ siteId } />
-				<QuerySiteFeatures siteIds={ [ siteId ] } />
-				<QueryWordadsStatus siteId={ siteId } />
-				{ notice }
-				{ component }
-			</div>
-		);
-	}
-}
-
-const mapStateToProps = ( state ) => {
-	const site = getSelectedSite( state );
-	const siteId = getSelectedSiteId( state );
-	const hasWordAdsFeature = siteHasWordAds( state, siteId );
-	const canActivateWordAds = canCurrentUser( state, siteId, 'activate_wordads' );
-
-	return {
-		site,
-		siteId,
-		siteSlug: getSelectedSiteSlug( state ),
-		canAccessAds: canAccessWordAds( state, siteId ),
-		canActivateWordadsInstant: ! site.options.wordads && canActivateWordAds && hasWordAdsFeature,
-		canManageOptions: canCurrentUser( state, siteId, 'manage_options' ),
```

Comment:
Most things are just moved from mapStatetoProps to top of functional component. But canManageOptions was not used in the component and has been deleted.